### ISSUE TITLE
yolov5 dataset format conversion

### ISSUE THREAD

**Issue by WangKeyi-Wky** [ultralytics/yolov5#13645]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

how to convert the formation of the dataset to the formation of yolov5?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WangKeyi-Wky, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ Tutorials to get started, where you can find quickstart guides for simple tasks like Custom Data Training all the way to advanced concepts like Hyperparameter Evolution:
- Tutorials: https://docs.ultralytics.com/yolov5/
- Custom Data Training: https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/
- Hyperparameter Evolution: https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/
This is an automated response 🤖 and an Ultralytics engineer will follow up to assist you soon.

If this is a 🐛 Bug Report, please provide a minimum reproducible example to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our Tips for Best Training Results: https://docs.ultralytics.com/guides/model-training-tips/

For dataset format/conversion questions specifically, please include:
-A small sample (2–3 images) with their current annotations
-The target data.yaml and label examples you want to achieve
-The exact commands or scripts you’ve tried
-Your environment details (OS, Python, PyTorch, CUDA) and YOLOv5 version
-Any error messages or screenshots

## Requirements
Python>=3.8.0 with all requirements.txt installed including PyTorch>=1.8. To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```
- Python: https://www.python.org/
- requirements.txt: https://github.com/ultralytics/yolov5/blob/master/requirements.txt
- PyTorch: https://pytorch.org/get-started/locally/

## Environments
YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python and PyTorch preinstalled):
- Notebooks with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See GCP Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/
- Amazon Deep Learning AMI. See AWS Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/
- Docker Image. See Docker Quickstart Guide: https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/ <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all YOLOv5 GitHub Actions Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 training, validation, inference, export and benchmarks on macOS, Windows, and Ubuntu every 24 hours and on every commit.
- Actions: https://github.com/ultralytics/yolov5/actions
- train.py: https://github.com/ultralytics/yolov5/blob/master/train.py
- val.py: https://github.com/ultralytics/yolov5/blob/master/val.py
- detect.py: https://github.com/ultralytics/yolov5/blob/master/detect.py
- export.py: https://github.com/ultralytics/yolov5/blob/master/export.py
- benchmarks.py: https://github.com/ultralytics/yolov5/blob/master/benchmarks.py

Thanks for sharing more details so we can help you faster 🙌

**Comment by pderrenger** [comment]
Hi @WangKeyi-Wky! YOLOv5 expects datasets in a specific format with images in folders and corresponding `.txt` label files containing normalized bounding box coordinates in the format `class x_center y_center width height`. 

You can check the [YOLOv5 Train Custom Data tutorial](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data) which explains the required dataset structure and includes examples for converting from common formats like COCO, VOC, and others. The tutorial also shows how to set up your `dataset.yaml` configuration file.

If you're converting from a specific format like COCO or VOC, there are conversion scripts available in the YOLOv5 repository under the `utils` folder that can help automate the process.
